### ISSUE TITLE
[Bug]: Dialog close announces unnecessary information

### ISSUE THREAD

**Issue by kpasagada** [microsoft/fluentui#23899]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz
    Memory: 4.13 GB / 15.79 GB
Browsers:
    Chrome: 103.0.5060.114
    Edge: Spartan (44.22000.120.0), Chromium (103.0.1264.49)
    Internet Explorer: 11.0.22000.120
Screen reader:
    NVDA version: 2022.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/kpasagada/pen/bGvBGva

### Bug Description

Dialog close announces all the screen reader announce messages. This is observed in NVDA + Chrome. 
This issue is also observed with Panels.
This issue DOES NOT happen in Fluent v7.

**To reproduce:**
1. Open the codepen and open the dialog by clicking the "Open Dialog" button.
2. Close the dialog using any button.
3. Observe screen reader announcements.

## Actual Behavior
The screen reader announces "do not announce this message" announcement.

## Expected Behavior
The screen reader should not announce anything unnecessary.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Thanks for filing this, but I was unable to reproduce the stated issue. Please, can you retest using the latest version of the lib and let us know if the issue still persists.

**Comment by kpasagada** [comment]
@gouttierre I am still able to reproduce this issue in the code pen. I am using NVDA + Chrome for this.

In the below recording, around 0:28, you can see that on dialog close, the screen reader announces "do not announce this message" when it is not supposed to.

https://user-images.githubusercontent.com/42693353/184942046-09d08257-2279-41dd-a7eb-3ee45fa957a4.mp4

**Comment by MurugeshSandari** [comment]
Hi Team,
Is the fix for this issue rolled out ? I see issue still persists in the latest version of fluent v8. Can you please provide update on this. 

https://user-images.githubusercontent.com/45966086/216769047-2caca8d6-69e3-48f8-9589-96f4e8bfd8f6.mp4
